### PR TITLE
[Draft][tflchef/circlechef] Use a dedicated type for custom operators

### PR DESCRIPTION
**Pull request by Seunghui98**
On going draft to use a dedicated type for custom operators in the recipe.

ONE-DCO-1.0-Signed-off-by: SeungHui Lee <shsh1004.lee@samsung.com>

---
Related Issue : https://github.com/Samsung/ONE/issues/11741

### CODE REVIEW

File: compiler/circlechef/core/src/ModelChef.cpp [hunk 1]
```diff
@@ -171,6 +177,12 @@ std::set<std::string> gather_customcode_set(const ::circlechef::ModelRecipe &mod
   std::set<std::string> customcode_set;
   for (const auto &operation : model_recipe.operation())
   {
+    if (operation.type() == "Custom")
+    {
+      customcode_set.insert(operation.custom_code());
+      continue;
+    }
```
Comment (mhs4670go) [inline compiler/circlechef/core/src/ModelChef.cpp:184]: These codes can substitute below codes.
```cxx
auto op_chef = op_chef_registry().lookup(operation.type()).create(&operation);
if (op_chef->code() == circle::BuiltinOperator_CUSTOM)
  customcode_set.insert(operation.type());
```

Which means above codes should be removed.

File: compiler/circlechef/core/src/ModelChef.cpp [hunk 2]
```diff
@@ -182,6 +194,12 @@ std::set<std::string> gather_customcode_set(const ::circlechef::ModelRecipe &mod
     const auto &graph = model_recipe.graph(g);
     for (const auto &operation : graph.operation())
     {
+      if (operation.type() == "Custom")
```
Comment (mhs4670go) [inline compiler/circlechef/core/src/ModelChef.cpp:197]: ditto

File: res/TensorFlowLiteRecipes/All_000/test.recipe [hunk 3]
```diff
@@ -18,13 +18,14 @@ operand {
   }
 }
 operation {
-  type: "All"
+  type: "Custom"
```
Comment (seanshpark) [inline res/TensorFlowLiteRecipes/All_000/test.recipe:21]: to land separate PRs for `compiler` and `res` files,
we need to introduce some intermediate changes
for example,
1/ in `res`, introduce `extype: "Custom"` and `custom_code: "All"`
2/ in `tflchef`, if `extype` exist, use that, if not use `type`
3/ in `res`, change `type` and remove `extype`
4/ in `tflchef` remove related with `extype`

Comment (Seunghui98) [reply res/TensorFlowLiteRecipes/All_000/test.recipe:21]: Thank you for your comment!! I will upload the code PR as you guided me. :)

File: compiler/tflchef/core/src/CustomOp/AddV2.cpp [hunk 4]
```diff
@@ -29,7 +29,11 @@ AddV2Chef::custom_value(flatbuffers::FlatBufferBuilder &fbb) const
 {
   auto &operation = (*_operation);
 
-  assert(operation.type() == "AddV2");
+  if (operation.has_extype() && operation.extype() == "Custom")
```
Comment (seanshpark) [inline compiler/tflchef/core/src/CustomOp/AddV2.cpp:32]: why not check `type() == `Custom` too ?

File: compiler/tflchef/core/src/CustomOp/AddV2.cpp [hunk 5]
```diff
@@ -29,7 +29,11 @@ AddV2Chef::custom_value(flatbuffers::FlatBufferBuilder &fbb) const
...
+  {
+    assert(operation.has_custom_code());
+    assert(operation.custom_code() == "AddV2");
+  }
```
Comment (seanshpark) [inline compiler/tflchef/core/src/CustomOp/AddV2.cpp:36]: as these blocks are repeated several times, would be better to add a method
